CMakeLists.txt

```diff
@@ -42,7 +42,9 @@ option(IREE_ENABLE_CPUINFO "Enables runtime use of cpuinfo for processor topolog
 option(IREE_BUILD_COMPILER "Builds the IREE compiler." ON)
 option(IREE_BUILD_TESTS "Builds IREE unit tests." ON)
 option(IREE_BUILD_BENCHMARKS "Builds IREE benchmark suites." OFF)
-option(IREE_ENABLE_COMPILATION_BENCHMARKS "Builds IREE benchmark suites for compilation statistics." OFF)
+# TODO(pzread): Currently IREE_ENABLE_COMPILATION_BENCHMARKS needs
+# IREE_BUILD_BENCHMARKS. Eventually we will decouple them.
+cmake_dependent_option(IREE_ENABLE_COMPILATION_BENCHMARKS "Builds IREE benchmark suites for compilation statistics." OFF ${IREE_BUILD_BENCHMARKS} OFF)
```

Comment:
(Sort of contradicting myself, but bear with me)

I'd be careful with complicated options and comments here, and these sorts of TODOs tend to stick around for a while, despite our good intentions. This is right near the top of the file and is the main place that people scanning through will look for what options they can set in the build. We might even want to move `IREE_BUILD_BENCHMARKS` to its own group, like how `IREE_BUILD_BINDINGS_TFLITE` is separate from the others down a few lines from here.

I think of it this way: people depending on IREE's CMake configuration downstream (e.g. users of IREE's C API, or people adding layers on top of the compiler) probably want to set a few high level options like `IREE_BUILD_COMPILER`, but they shouldn't need to wade through a bunch of options that are primarily for people working on IREE itself.

Reply:
I'll actively work on refactoring benchmark suite for a while, so the TODO is more like a note for me. I'll try to address soon.

Separated the benchmark suite options into its own section.